src/svc_dg.c

```diff
@@ -415,13 +417,29 @@ svc_dg_reply(struct svc_req *req)
 	msg->msg_iov = &iov;
 	msg->msg_iovlen = 1;
 	msg->msg_name = (struct sockaddr *)&xprt->xp_remote.ss;
-	msg->msg_namelen = xprt->xp_remote.nb.len;
-	/* cmsg already set in svc_dg_rendezvous */
+	msg->msg_namelen = sizeof(struct sockaddr_storage);
+	msg->msg_control = msg_control;
+	msg->msg_controllen = sizeof(msg_control);
+	msg->msg_flags = 0;
+
+	cmsg = CMSG_FIRSTHDR(msg);
+	cmsg->cmsg_level = (xprt->xp_local.ss.ss_family == AF_INET)
+		? IPPROTO_IP : IPPROTO_IPV6; /* a.k.a. SOL_IP and SOL_IPV6 */
+	cmsg->cmsg_type = (xprt->xp_local.ss.ss_family == AF_INET)
+		? IP_PKTINFO : IPV6_PKTINFO;
+	*(struct in_pktinfo*)CMSG_DATA(cmsg) =
```

Comment:
I don't think this will work for IPv6, as in6_pktinfo is larger than in_pktinfo.  Maybe memcpy() with the computed length below?